proposals/threads/Globals.md

```diff
@@ -2,6 +2,316 @@
 
 This page describes a proposal for importing and exporting mutable globals.
 
+## Rationale
+
+Without the ability to import and export mutable globals, it is inconvenient to
+provide mutable thread-local values that can be dynamically linked, such as the
+C++ stack pointer (SP).
+
+### Example: Dynamically Linking SP with Single Agent
+
+Let's assume we have two modules that are dynamically linked, `m1` and
+`m2`. They both use the C++ stack pointer. In the MVP, it's not possible to
+import or export a mutable global. In a single-agent program, we can work
+around this by storing the stack pointer in linear memory:
+
+```
+(module $m1
+  (memory (export "memory") 1)
+  ;; Address 0 is the address of SP
+  ;; The stack starts at 0x100
+  (data (i32.const 0) "\00\01\00\00)
+
+  (func
+    ;; SP = SP + 64
+    (i32.store
+      (i32.const 0)  ;; SP
+      (i32.add
+        (i32.load (i32.const 0))  ;; SP
+        (i32.const 64)))
+    ...
+  )
+  ...
+)
+
+(module $m2
+  (import "env" "memory" (memory 1))
+  ...
+)
+```
+
+Then the modules can be instantiated as follows:
+
+```
+WebAssembly.instantiate(m1Bytes, {}).then(
+    ({_, instance}) => {
```

Comment:
Here and elsewhere, it should just be `({instance})`; no `_,` necessary. (That tries to destructure a non-existant property `_` from the result, so `_` is always `undefined`.)